libkbfs/bserver_remote.go

```diff
@@ -27,11 +26,18 @@ const (
 	BServerTokenExpireIn = 2 * 60 * 60 // 2 hours
 )
 
+type blockServerRemoteConfig interface {
+	diskBlockCacheGetter
+	codecGetter
+	signerGetter
+	currentInfoGetterGetter
+	logMaker
+}
+
 // BlockServerRemote implements the BlockServer interface and
 // represents a remote KBFS block server.
 type BlockServerRemote struct {
-	codec      kbfscodec.Codec
-	cig        currentInfoGetter
+	config     blockServerRemoteConfig
```

Comment:
I don't think it's desirable to design classes around `Config`; in a perfect world, we wouldn't have `Config` at all, so why design around it? The current pattern of having `fooConfig` structs/interfaces is only to prevent every class from depending on every other class through `Config` without having to rewrite the world. If a type is already taking a few parameters directly, I see no reason to convert it to take a `fooConfig` type for the sake of a false economy.

(I'll elaborate in my other reply to your other comment below.)